### ISSUE TITLE
FP S2814 for TypeScript functions, classes, enums, and namespaces

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#1547]
See #1502 

We can fix the false positives by setting [ignoreDeclarationMerge](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-redeclare.md#ignoredeclarationmerge) option to true .

**Comment by vilchik-elena** [comment]
```
namespace X {
// ...
}
namespace X { // FP
//...
}
```

```
class A {
}
namespace A { // FP
}
```
